app/controllers/activities_controller.rb

```diff
@@ -0,0 +1,18 @@
+class ActivitiesController < ApplicationController
+
+  before_action :set_project
```

Comment:
Set the @project IV directly on the action. That makes the code more readable meanwhile it grows up. Use this kind of hook only when is strictly necessary.